### PR TITLE
Fix: adding socket script and social util script in Delayed or on the need basis

### PR DESCRIPTION
**Pull request by anagarwa**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/research/equity
- After: https://socket-delayed--icicidirect--aemsites.hlx.live/research/equity

- [ ] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation
------------- | -------------
 For e.g. carousel | [doc-link](https://main--icicidirect--aemsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/carousel&index=0)

### CODE REVIEW

**Comment by ravverma** [comment]
One change i have noticed that it load the header block first so it showed the sensex from API with data and time and then when feed started the date disappeared.

https://github.com/aemsites/icicidirect/assets/46780200/04f20302-7445-4ee9-a637-28d638762917

File: scripts/aem.js [hunk 1]
```diff
@@ -646,7 +646,7 @@ async function loadHeader(header) {
   const headerBlock = buildBlock('header', '');
   header.append(headerBlock);
   decorateBlock(headerBlock);
-  return loadBlock(headerBlock);
+  await loadBlock(headerBlock);
```
Comment (sdmcraft) [inline scripts/aem.js:649]: This should also return the loaded block.

File: scripts/scripts.js [hunk 2]
```diff
@@ -107,15 +106,6 @@ function decorateAnchors(element = document) {
...
   }
-
-  const socialShareAnchors = anchors.filter((anchor) => {
-    const img = anchor.querySelector('img');
-    return img && img.src.includes('/icons/gray-share-icon.svg');
-  });
-
-  if (socialShareAnchors.length > 0) {
-    decorateSocialShare(socialShareAnchors);
-  }
```
Comment (sdmcraft) [inline scripts/scripts.js:118]: Is this no longer needed? Can't find it moved to anywhere else.

Comment (anagarwa) [reply scripts/scripts.js:118]: @sdmcraft  correct, in eager phase we do not have social button. and buttons loaded in lazy phase are decorated already

File: scripts/aem.js (hunk 1 above)
Comment (anagarwa) [reply scripts/aem.js:649]: @sdmcraft  this is already part of https://github.com/aemsites/icicidirect/pull/291 so I will remove this once https://github.com/aemsites/icicidirect/pull/291 gets merged.

**Comment by anagarwa** [comment]
> One change i have noticed that it load the header block first so it showed the sensex from API with data and time and then when feed started the date disappeared.
> 
>  Screen.Recording.2024-06-25.at.14.28.52.mov

@ravverma thanks for pointing, let me check this

**Review by sdmcraft — APPROVED** [review]
Approving. 
I assume we addressed https://github.com/aemsites/icicidirect/pull/294#issuecomment-2188626656 (can't check as market is closed)

**Comment by anagarwa** [comment]
@sdmcraft  let me add a issue to check again tomorrow when market is open to see the behaviour change in main and current branch.

**Comment by anagarwa** [comment]
> > One change i have noticed that it load the header block first so it showed the sensex from API with data and time and then when feed started the date disappeared.
> > Screen.Recording.2024-06-25.at.14.28.52.mov
> 
> @ravverma thanks for pointing, let me check this

seems to be fix now, can you please check ?

**Comment by anagarwa** [comment]
> @sdmcraft let me add a issue to check again tomorrow when market is open to see the behaviour change in main and current branch.

Open the issue https://github.com/aemsites/icicidirect/issues/295 to further confirm when market opens

**Pull request merged**